test/core/event_engine/thread_pool_test.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/lib/event_engine/thread_pool.h"
+
+#include <atomic>
+#include <chrono>
+#include <thread>
+
+#include <gtest/gtest.h>
+
+#include "absl/synchronization/notification.h"
+#include "gtest/gtest.h"
+
+#include <grpc/support/log.h>
+
+namespace grpc_event_engine {
+namespace experimental {
+
+TEST(ThreadPoolTest, CanRunClosure) {
+  ThreadPool p(1);
+  absl::Notification n;
+  p.Add([&n] { n.Notify(); });
+  n.WaitForNotification();
+}
+
+TEST(ThreadPoolTest, CanDestroyInsideClosure) {
+  auto p = std::make_shared<ThreadPool>(1);
+  absl::Notification n;
```

Comment:
I don't believe we can safely use absl::Notification on all platforms yet.

Reply:
now uses our polyfill